synapse/storage/databases/main/transactions.py

```diff
@@ -218,6 +218,7 @@ def _set_destination_retry_timings(
                         retry_interval = EXCLUDED.retry_interval
                     WHERE
                         EXCLUDED.retry_interval = 0
+                        OR destinations.retry_interval IS NULL
```

Comment:
How is this check related?

Reply:
This is analogous to

> `prev_row["retry_interval"] is None`

but in the native upsert form.